src/Uno.UI/UI/Xaml/VisualStateManager.cs

```diff
@@ -112,6 +112,19 @@ public static bool GoToState(Control control, string stateName, bool useTransiti
 				return false;
 			}
 
+			if(templateRoot is FrameworkElement fe)
+			{
+				if(fe.GoToElementState(stateName, useTransitions))
```

Comment:
You can reduce nesting:

```suggestion
				if(if(templateRoot is FrameworkElement fe
					 && fe.GoToElementState(stateName, useTransitions))
```